x-pack/plugins/alerting/public/hooks/use_find_maintenance_windows.ts

```diff
@@ -36,14 +47,10 @@ export const useFindMaintenanceWindows = (props?: UseFindMaintenanceWindowsProps
     }
   };
 
-  const {
-    isLoading,
-    isFetching,
-    isInitialLoading,
-    data = [],
-    refetch,
-  } = useQuery({
-    queryKey: ['findMaintenanceWindows'],
+  const queryKey = ['findMaintenanceWindows', page, perPage, search, selectedStatus];
+
+  const { isLoading, isFetching, isInitialLoading, data, refetch } = useQuery({
```

Comment:
Let's also add `keepPreviousData: true` to avoid having the table clear every time we change status.